From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brandon Jones <bajones@chromium.org>
Date: Fri, 14 Aug 2026 12:53:44 -0700
Subject: [M150] Validate framebufferTextureMultiviewOVR target, attachment

Original change's description:
> Validate framebufferTextureMultiviewOVR target, attachment
>
> Previously these were being passed through unvalidated, which could
> lead to resource tracking issues if an out-of-range value was
> given. While the extension definitions for this method don't state
> any explicit validation to perform on these values it should be
> safe to assume they should be treated like values being passed to
> framebufferTexture2D, and thus the same validation is reused here.
>
> Fixed: 543082390
> Change-Id: I63a172b6723e6e5a626d50583668b4dc0ca2a33a
> Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8243727
> Commit-Queue: Kai Ninomiya <kainino@chromium.org>
> Reviewed-by: Kai Ninomiya <kainino@chromium.org>
> Cr-Commit-Position: refs/heads/main@{#1677709}

(cherry picked from commit 2b3a653dda35893442fbda35c6d8f89b0c683002)

Bug: 546214453,543082390
Change-Id: I63a172b6723e6e5a626d50583668b4dc0ca2a33a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/8260833
Commit-Queue: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com <chrome-cherry-picker@chops-service-accounts.iam.gserviceaccount.com>
Bot-Commit: rubber-stamper@appspot.gserviceaccount.com <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/branch-heads/7871@{#4295}
Cr-Branched-From: f542126b8c1b3e80104b26bb05ec830bd1206f29-refs/heads/main@{#1639810}

diff --git a/third_party/blink/renderer/modules/webgl/ovr_multiview_2.cc b/third_party/blink/renderer/modules/webgl/ovr_multiview_2.cc
index 8ecc9f9d3f81cbaf3b8be1dba578328086cca491..6dec060f399e928ce83185d7329fcb60074542de 100644
--- a/third_party/blink/renderer/modules/webgl/ovr_multiview_2.cc
+++ b/third_party/blink/renderer/modules/webgl/ovr_multiview_2.cc
@@ -61,6 +61,11 @@ void OVRMultiview2::framebufferTextureMultiviewOVR(GLenum target,
              ->ValidateTexFuncLayer("framebufferTextureMultiviewOVR", textarget,
                                     base_view_index + num_views - 1))
       return;
+    if (!static_cast<WebGL2RenderingContextBase*>(scoped.Context())
+             ->ValidateFramebufferFuncParameters(
+                 "framebufferTextureMultiviewOVR", target, attachment)) {
+      return;
+    }
     if (!scoped.Context()->ValidateTexFuncLevel(
             "framebufferTextureMultiviewOVR", textarget, level))
       return;
